TrackingPerformance/Condor/condorJobs_sim.py

```diff
@@ -1,156 +1,248 @@
-
-# ====================
-# Submit Job to Condor
-# ====================
-os.system("cd "+ directory_jobs + "; condor_submit condor_script.sub")
+        sys.exit(1)
+
+    SIMEOSDir.mkdir(
+        parents=True, exist_ok=True
+    )  # This will create the directory if it doesn't exist, without raising an error if it does
+
+    # =======================
+    # Simulation Job Creation
+    # =======================
+
+    # Create all possible combinations
+    import itertools
+
+    iter_of_combined_variables = itertools.product(
+        thetaList_, momentumList_, particleList_, DetectorModelList_
+    )
+
+    need_to_create_scripts = False
+
+    for theta, momentum, part, dect in iter_of_combined_variables:
+        for task_index in range(N_jobs_per_para_set):
+
+            output_file_name_parts = [
+                f"SIM_{dect}",
+                f"{part}",
+                f"{theta}_deg",
+                f"{momentum}_GeV",
+                f"{Nevt_per_job}_evts",
+                f"{task_index}",
+            ]
+            output_file_name = Path("_".join(output_file_name_parts)).with_suffix(
+                ".edm4hep.root"
+            )
 
+            # Check if the output file already exists and has correct Nb of events
+            output_dir = SIMEOSDir / part
+            output_dir.mkdir(parents=True, exist_ok=True)
+            output_file_path = output_dir / output_file_name
+            if CHECK_OUTPUT and output_file_path.exists():
+                root_file = ROOT.TFile(output_file_path, "READ")
+                events_tree = root_file.Get("events")
+                if events_tree:  # FIXME: why no else?
+                    if events_tree.GetEntries() == int(Nevt_per_job):
+                        root_file.Close()
+                        continue
+                root_file.Close()
+            else:
+                need_to_create_scripts = True
+
+            # if len(DetectorModelList_) != 1 or DetectorModelList_[0] != "ILD_l5_v11":
+            #     raise ValueError("so far only ILD_l5_v11 possible")
+
+            # Build ddsim command
+            arguments = [
+                # f" --compactFile /afs/cern.ch/work/g/gasadows/k4geo/FCCee/CLD/compact/{DetectorModelList_[0]}_3T/{DetectorModelList_[0]}.xml "
+                f" --compactFile $k4geo_DIR/{config.detModPaths['ILD_l5_v11']}",  # Note the change to use double quotes for the dictionary key
+                f"--outputFile {output_file_name}",
+                f"--steeringFile {config.sim_steering_file}",  # "CLDConfig/CLDConfig/cld_steer.py "
+                # TODO: FIX
+                # "--enableGun",
+                # f"--gun.particle {part}-",
+                # f"--gun.energy {momentum}*GeV",
+                # "--gun.distribution uniform",
+                # f"--gun.thetaMin {theta}*deg",
+                # f"--gun.thetaMax {theta}*deg",
+                # "--crossingAngleBoost 0",
+                # f"--numberOfEvents {Nevt_per_job}",
```

Comment:
Does something break when commenting it in? Without it the file is very useless :)

Reply:
done